llama_hub/docstring_walker/README.md

```diff
@@ -0,0 +1,133 @@
+print(example1_docs[0].text[:500])
+
+>> """ 
+Module name: base 
+Docstring: None
+Class name: DocstringWalker 
+Docstring: A loader for docstring extraction and building structured documents from them.
+Recursively walks a directory and extracts docstrings from each Python module - starting from the module
+itself, then classes, then functions. Builds a graph of dependencies between the extracted docstrings.
+
+Function name: load_data, In: DocstringWalker 
+Docstring: Load data from the specified code directory.
+Additionally, after loading t
+ """
+
+ # Step 3: Feed documents into Llama Index
+ example1_index = VectorStoreIndex(example1_docs, service_context=service_context)
+
+ # Step 4: Query the index
+ example1_qe = example1_index.as_query_engine(service_context=service_context)
+
+
+# Step 5: And start querying the index
+print(example1_qe.query("What are the main functions used by DocstringWalker? Describe each one in points.").response)
+
+>> """
+1. load_data: This function loads data from a specified code directory and builds a dependency graph between the loaded documents. The graph is stored as an attribute of the class.
+
+2. process_directory: This function processes a directory and extracts information from Python files. It returns a tuple containing a list of Document objects and a networkx Graph object. The Document objects represent the extracted information from Python files, and the Graph object represents the dependency graph between the extracted documents.
+
+3. read_module_text: This function reads the text of a Python module given its path and returns the text of the module.
+
+4. parse_module: This function parses a single Python module and returns a Document object with extracted information from the module.
+
+5. process_class: This function processes a class node in the AST and adds relevant information to the graph. It returns a string representation of the processed class node and its sub-elements.
+
+6. process_function: This function processes a function node in the AST and adds it to the graph. It returns a string representation of the processed function node with its sub-elements.
+
+7. process_elem: This is a generic function that processes an element in the abstract syntax tree (AST) and delegates the execution to more specific functions based on the type of the element. It returns the result of processing the element.
+"""
+```
+
+
+# Example 2 - check some arbitrarily selected module
+
+Now we can check how to apply DocstringWalker to some files under an arbitrary directory. Let's use the code from the PyTorchGeometric KGE (Knowledge Graphs Embedding) directory. 
+You can find its original documentation and classes here: https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#kge-models
+
+We import the module and use its filepath directly.
+
+```python
+import os
+from torch_geometric.nn import kge
+
+# Step 1 - get path to module
+module_path = kge_dir = os.path.dirname(kge.__file__)
```

Comment:
nit: why name this two different variables?

Reply:
Clearly a mistake, will correct this, sorry.